src/main/java/com/google/javascript/gents/TypeAnnotationPass.java

```diff
@@ -199,14 +199,19 @@ public void visit(NodeTraversal t, Node n, Node parent) {
           // creating each MEMBER_VARIABLE_DEFs so code generator works.
           if (bestJSDocInfo != null && bestJSDocInfo.hasTypedefType()) {
             Node typedefTypeRoot = bestJSDocInfo.getTypedefType().getRoot();
-            //LC
-            //    LB
-            //        COLON
-            //            STRING_KEY a
-            //            STRING number
-            //        COLON
-            //            STRING_KEY b
-            //            STRING number
+            if ((typedefTypeRoot.getToken() == Token.BANG)
+                || (typedefTypeRoot.getToken() == Token.QMARK)) {
+              typedefTypeRoot = typedefTypeRoot.getFirstChild();
+            }
+            //(BANG|QMASK)
+            //    LC
+            //        LB
```

Comment:
what if I have a typedef that's a simple alias?

```js
/** @typedef {string} */
var MyStringType;
```

In general, a typedef can contain any arbitrary type JSDoc tree – we must be careful that we actually handle those trees. If it is an invariant guaranteed by other code that there can only be member defs here, we should at least add a comment pointing to the code, or better try to move the code together in one location.